RevenueCat/Scripts/Purchases.cs

```diff
@@ -88,7 +101,16 @@ private void Start()
     // Call this if you want to reset with a new user id
     private void Setup(string newUserId)
     {
-        _wrapper.Setup(gameObject.name, revenueCatAPIKey, newUserId, observerMode, userDefaultsSuiteName);
+        var apiKey = deprecatedLegacyRevenueCatAPIKey;
+        if (String.IsNullOrEmpty(apiKey))
+        {
+            if (Application.platform == RuntimePlatform.IPhonePlayer)
```

Comment:
I wonder if this should take precedence over `deprecatedLegacyRevenueCatAPIKey`? So only check that iff `revenueCatAPIKeyIOS` or `revenueCatAPIKeyAndroid` aren't defined?